### ISSUE TITLE
CT noise addition, when inside a training loop, stores gradients

### ISSUE THREAD

**Issue by AnderBiguri** [CambridgeCIA/LION#138]
We should add a optional input argument `enable_gradients=False` to the noise simulator.
